client_plugin/drivers/shared/etcdops.go

```diff
@@ -0,0 +1,269 @@
+				"members len": len(lresp.Members)},
+		).Error("Failed to list member for ETCD")
+		return err
+	}
+
+	peerAddr := etcdScheme + nodeAddr + etcdPeerPort
+	existing := false
+	for _, member := range lresp.Members {
+		if member.PeerURLs[0] == peerAddr {
+			if member.Name == "" {
+				// already added, not started
+				log.WithFields(
+					log.Fields{"nodeID": nodeID,
+						"peerAddr": peerAddr},
+				).Info("This node is already added as a etcd member, but not started yet ")
+
+				existing = true
+			} else {
+				// already added and started, need to reset
+				log.WithFields(
+					log.Fields{"nodeID": nodeID,
+						"peerAddr": peerAddr},
+				).Info("This node is already added as a etcd member and started, need to remove first ")
+
+				_, err = cli.MemberRemove(context.Background(), member.ID)
+				if err != nil {
+					log.WithFields(
+						log.Fields{"peerAddr": peerAddr,
+							"member.ID": member.ID},
+					).Error("Failed to remove member for ETCD")
+					return err
+				}
+			}
+			break
+		}
+	}
+
+	initCluster := ""
+	if !existing {
+		peerAddrs := []string{peerAddr}
+		aresp, err := cli.MemberAdd(context.Background(), peerAddrs)
+		if err != nil {
+			log.WithFields(
+				log.Fields{"leaderAddr": leaderAddr,
+					"error":       err,
+					"members len": len(aresp.Members)},
+			).Error("Failed to add member for ETCD")
+			return err
+		}
+		for _, member := range aresp.Members {
+			if member.Name != "" {
+				initCluster += member.Name + "=" + member.PeerURLs[0] + ","
+			}
+		}
+	} else {
+		for _, member := range lresp.Members {
+			if member.Name != "" {
+				initCluster += member.Name + "=" + member.PeerURLs[0] + ","
+			}
+		}
+	}
+
+	lines := []string{
+		"--name", nodeID,
+		"--advertise-client-urls", etcdScheme + nodeAddr + etcdClientPort,
+		"--initial-advertise-peer-urls", etcdScheme + nodeAddr + etcdPeerPort,
+		"--listen-client-urls", etcdScheme + etcdListenURL + etcdClientPort,
+		"--listen-peer-urls", etcdScheme + etcdListenURL + etcdPeerPort,
+		"--initial-cluster-token", etcdClusterToken,
+		"--initial-cluster", initCluster + nodeID + "=" + etcdScheme + nodeAddr + etcdPeerPort,
+		"--initial-cluster-state", etcdClusterStateExisting,
+	}
+
+	go etcdService(lines)
+
+	return nil
+}
+
+func etcdService(cmd []string) {
+	_, err := exec.Command("/bin/etcd", cmd...).Output()
+	if err != nil {
+		log.WithFields(
+			log.Fields{"error": err, "cmd": cmd},
+		).Error("Failed to start etcd command ")
+	}
+}
+
+func (d *VolumeDriver) createEtcdClient() *etcdClient.Client {
+	dclient := d.dockerd
+
+	info, err := dclient.Info(context.Background())
+	if err != nil {
+		log.WithFields(
+			log.Fields{"error": err},
+		).Error("Failed to get Info from docker client ")
+		return nil
+	}
+
+	for _, manager := range info.Swarm.RemoteManagers {
+		cli, err := addrToEtcdClient(manager.Addr)
+		if err == nil {
+			return cli
+		}
+	}
+
+	log.Errorf("Failed to create etcd client according to manager info")
```

Comment:
Need to print extra inffo here; if we got here it's likely there was no debug print yet